### DISCUSSION TITLE
Application not auto-starting properly

### DISCUSSION THREAD

**Discussion by stevalkr** [i3/i3#5640] [Q&A]
Hi, I have following in config:
```
# background apps
exec --no-startup-id fcitx5
exec --no-startup-id pipewire
exec --no-startup-id flameshot
exec --no-startup-id 1password --silent
exec --no-startup-id Telegram -startintray

# workspace-1
exec --no-startup-id i3-msg "workspace 1; append_layout $HOME/.config/i3/workspace-1.json"
exec --no-startup-id vivaldi-stable --gtk-version=4
exec --no-startup-id thunderbird
exec --no-startup-id kitty
```

All components except 1password and Telegram function seamlessly. On occasion, one of the latter two may be absent from the system tray, yet their processes remain accessible through the `ps -ef` command. Any ideas?

**Comment by okraits** [comment]
Do those two programs start up correctly if you run them from a terminal? And why do you disable the startup id for all of those programs?

**Reply by stevalkr** [reply]
1. Yes, `--silent` and `-startintray` is to have them start in tray. They aren't always not showing in the tray, sometimes it works, sometimes not.
2. According to user guide, 
   > The --no-startup-id parameter disables startup-notification support for this particular exec command. With startup-notification, i3 can make sure that a window appears on the workspace on which you used the exec command. Also, it will change the X11 cursor to watch (a clock) while the application is launching. So, if an application is not startup-notification aware (most GTK and Qt using applications seem to be, though), you will end up with a watch cursor for 60 seconds.
   
   I use this parameter to avoid cursor change

**Reply by okraits** [reply]
If running them from a terminal doesn't work reliably either, this indicates that there's some issue with these programs, independent of i3. I would suggest to consult the support of those programs.

**Reply by stevalkr** [reply]
Sorry for my poor English, I mean "Yes, they start up correctly if I run them from a terminal, They aren't always not showing in the tray when started from i3"
